frontend/components/Message.tsx

```diff
@@ -1,9 +1,57 @@
 import { getNickCSSClass } from '../data/UserHash';
+import { hasTrailingChar, splitToTextWithoutTrailingAndTrailing } from '../util/stringUtils';
 
 export default function Message(props: IMessage): JSX.Element {
 
   /**
-   * Returns a message formatted in the way irssi formats it for /me
+   * Wraps an a href around text, given an url and clickable text.
+   */
+  function formatLinkNormally(href: string, clickable: string): JSX.Element {
+    return <span> <a href={href}>{clickable}</a></span>;
+  }
+
+  /**
+   * 
+   * Wraps an a href around text, given an url and clickable text,
+   * but excludes a trailing , or . from the clickable link.
+   */
+  function formatLinkWithEndChar(href: string, clickable: string, endChar: string): JSX.Element {
+    return <span> <a href={href}>{clickable}</a>{endChar}</span>;
+  }
+
+  /**
+   * Calls either link formatting method depending on whether the href or clickable text contains trailing characters.
+   * Also splits up the href and clickable into parts with and without the trailing characters.
+   */
+  function formatLink(href: string, clickable: string): JSX.Element {
+    if (hasTrailingChar(clickable) || hasTrailingChar(href)) {
+      const [clickableWithoutEndChar, endChar] = splitToTextWithoutTrailingAndTrailing(clickable);
+      const hrefWithoutEndChar = splitToTextWithoutTrailingAndTrailing(href)[0];
+      return formatLinkWithEndChar(hrefWithoutEndChar, clickableWithoutEndChar, endChar);
+    }
+    return formatLinkNormally(href, clickable);
+  }
+
+  /** 
+   * If a message contains URLs or (/)r/<subreddit> those parts will be linked.
+   */
+  function linkifyMessageContent(message: string): JSX.Element[] {
```

Comment:
Is the function run on every line? Could it be better if it's only run on lines that actually contain links?

I'm not sure about the performance impact but I smell more potential for funky shit to happen if it just splits all the messages.

Reply:
This method is now only run on messages that contains parts that are to be linked.